src/Microsoft.AspNet.Authorization/DelegateRequirement.cs

```diff
@@ -0,0 +1,22 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+
+namespace Microsoft.AspNet.Authorization
+{
+    public class DelegateRequirement : AuthorizationHandler<DelegateRequirement>, IAuthorizationRequirement
+    {
+        public Action<AuthorizationContext, DelegateRequirement> Handler { get; }
+
+        public DelegateRequirement(Action<AuthorizationContext, DelegateRequirement> handleMe)
+        {
+            Handler = handleMe;
+        }
+
+        protected override void Handle(AuthorizationContext context, DelegateRequirement requirement)
+        {
+            Handler.Invoke(context, requirement);
+        }
+    }
```

Comment:
We'll keep delegate as sync only, because its only supposed to take those two inputs, and is unable to use injection